app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -40,14 +40,13 @@ register_host() {
        --data "host[name]=$(hostname --fqdn)" \
        --data "host[build]=false" \
        --data "host[managed]=false" \
-       <%= "--data \"host[organization_id]=#{@organization.id}\"" if @organization -%>
-       <%= "--data \"host[location_id]=#{@location.id}\"" if @location -%>
-       <%= "--data \"host[hostgroup_id]=#{@hostgroup.id}\"" if @hostgroup -%>
-       <%= "--data \"host[operatingsystem_id]=#{@operatingsystem.id}\"" if @operatingsystem -%>
-       <%= "--data \"setup_insights=#{@setup_insights}\"" unless @setup_insights.nil? -%>
-       <%= "--data \"setup_remote_execution=#{@setup_remote_execution}\"" unless @setup_remote_execution.nil? -%>
-       <%= "--data \"remote_execution_interface=#{@remote_execution_interface}\"" if @remote_execution_interface.present? -%>
-
+<%= "       --data 'host[organization_id]=#{@organization.id}' \\\n" if @organization -%>
+<%= "       --data 'host[location_id]=#{@location.id}' \\\n" if @location -%>
+<%= "       --data 'host[hostgroup_id]=#{@hostgroup.id}' \\\n" if @hostgroup -%>
+<%= "       --data 'host[operatingsystem_id]=#{@operatingsystem.id}' \\\n" if @operatingsystem -%>
+<%= "       --data 'setup_insights=#{@setup_insights}' \\\n" unless @setup_insights.nil? -%>
+<%= "       --data 'setup_remote_execution=#{@setup_remote_execution}' \\\n" unless @setup_remote_execution.nil? -%>
+<%= "       --data 'remote_execution_interface=#{@remote_execution_interface}' \\\n" if @remote_execution_interface.present? -%>
```

Comment:
You still end with `\\\n` here and I'm not sure if that works well. I'd still suggest to have these dynamic lines before static ones.

Reply:
I am sure it works since I tested it, here's the rendered output that you can try running in your terminal

```sh
register_host() {
  curl --cacert $SSL_CA_CERT --request POST https://foreman.example.tst/register \
       -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.eyJ1c2VyX.....k' \
       --data "host[name]=$(hostname --fqdn)" \
       --data "host[build]=false" \
       --data "host[managed]=false" \
       --data 'host[organization_id]=2' \
       --data 'host[location_id]=1' \
       --data 'host[operatingsystem_id]=5' \
       --data 'setup_insights=true' \
}

```

I don't like moving this above the static lines, static data lines indicates these always need to be set and therefore are first. Besides that, the following code looks even more weird to me


```erb
register_host() {
  curl --cacert $SSL_CA_CERT --request POST <%= @registration_url %> \
       <%= headers.join(' ') %> \
<%= "       --data 'host[organization_id]=#{@organization.id}' \\\n" if @organization -%>
<%= "       --data 'host[location_id]=#{@location.id}' \\\n" if @location -%>
<%= "       --data 'host[hostgroup_id]=#{@hostgroup.id}' \\\n" if @hostgroup -%>
<%= "       --data 'host[operatingsystem_id]=#{@operatingsystem.id}' \\\n" if @operatingsystem -%>
<%= "       --data 'setup_insights=#{@setup_insights}' \\\n" unless @setup_insights.nil? -%>
<%= "       --data 'setup_remote_execution=#{@setup_remote_execution}' \\\n" unless @setup_remote_execution.nil? -%>
<%= "       --data 'remote_execution_interface=#{@remote_execution_interface}' \\\n" if @remote_execution_interface.present? -%>
       --data "host[name]=$(hostname --fqdn)" \
       --data "host[build]=false" \
       --data "host[managed]=false"
}
```